### PR TITLE
Fixed an issue #26724  with added category to cart price rule by clic…

### PR DESCRIPTION
**Pull request by bruvinsky**
### Description (*)
Fixed an issue #26724 with added category to cart price rule by clicking on checkbox

### Fixed Issues (if relevant)

1. magento/magento2#26724: Unable add category to cart price rule by clicking on checkbox

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by hws47a** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @hws47a. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @hws47a, here is your [new Magento instance](https://pr-26791.instances.magento-community.engineering).
Admin access: [https://pr-26791.instances.magento-community.engineering/admin_d8c6](https://pr-26791.instances.magento-community.engineering/admin_d8c6)
Login: `fe4da5d5` Password: `5352867199c7`
Instance will be terminated in up to 3 hours.

**Comment by hws47a** [comment]
The issue seems fixed 👍

**Comment by hws47a** [comment]
Hi @bruvinsky, it is recommended to cover all of the fixes with tests. 
In this case, it could be the MFTF test.
You can also check if there is an existing test that doesn't cover the case or it is false-positive.

**Comment by hws47a** [comment]
All tests have passed except unrelated failures of functional tests

**Review by markshust — DISMISSED** [review]
Confirmed the fix works for me, code looks good

**Review by hws47a — CHANGES_REQUESTED** [review]
It needs to be covered by tests: unit or MFTF will fit here

File: dev/tests/integration/testsuite/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/NewActionHtmlTest.php [hunk 1]
```diff
@@ -31,6 +31,11 @@ class NewActionHtmlTest extends AbstractBackendController
      */
     private $formName = 'test_form';
 
+    /**
+     * @var string
+     */
+    private $jsFromObject = 'form';
```
Comment (hostep) [inline dev/tests/integration/testsuite/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/NewActionHtmlTest.php]: @Nazar65: there's a typo, should be `jsFormObject`, not `jsFromObject` 😉

Comment (engcom-Golf) [reply dev/tests/integration/testsuite/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/NewActionHtmlTest.php]: Nice catch :+1:  thank you

**Review by hws47a — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @hws47a, thank you for the review.
[ENGCOM-7111](https://jira.corp.magento.com/browse/ENGCOM-7111) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Before:**
Screenshot from 2020-03-17 11-32-46

 **After:**
after

**Comment by engcom-Golf** [comment]
Failed functional tests not related to the changes in this PR.

**Review by Nazar65 — CHANGES_REQUESTED** [review]
there was is one merge conflict, and looks like on mainline someone already fixed this issue, need to re review

**Comment by slavvka** [comment]
There were test failures. I don't know how it could get approved

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @bruvinsky. Looks like the issue mentioned in PR is already fixed. Can you please verify it?

**Comment by bruvinsky** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @bruvinsky. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by engcom-Echo** [comment]
Hi @bruvinsky , I'm closing this PR now due to inactivity, https://github.com/magento/magento2/pull/26791#pullrequestreview-432627638
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**